magefiles/scripts.go

```diff
@@ -69,6 +77,61 @@ func getGitInfo() (*gitInfo, error) {
 	}, nil
 }
 
+func getLatestVersion() (string, error) {
+	version, err := getLatestVersionGit()
+	if err != nil {
+		return "", err
+	}
+	if version != "" {
+		return version, nil
+	}
+	fmt.Println("Unable to determine version from git tags. Using changelog.")
+
+	version, err = getLatestVersionChangelog()
+	if err != nil {
+		return "", err
+	}
+	if version != "" {
+		return version, nil
+	}
+	fmt.Println("Unable to determine version the changelog. Using 0.0.0")
+
+	return "0.0.0", nil
+}
+
+func getLatestVersionGit() (string, error) {
+	version, err := sh.Output("git", "describe", "--tags", "--abbrev=0")
+	if err != nil && !strings.Contains(err.Error(), "exit code 128") {
+		return "", err
+	}
+	return version, nil
+}
+
+func getLatestVersionChangelog() (string, error) {
+	changelogs, err := os.ReadDir("contrib/changelog/prod")
+	if err != nil {
+		return "", err
+	}
+	changelog := slices.MaxFunc(changelogs, func(a fs.DirEntry, b fs.DirEntry) int {
```

Comment:
nit: slices.MaxFunc panics if `changelogs` is empty. Even though it should never happen, what about checking its length, and returning early if actually empty?

Reply:
Another nit, sorry!
Come to think about it and maybe we can leverage other semver package, it got `Sort` method that consumes a slice of strings, thus we could get rid of this temporary conversion a string to a semver. Sort thingies, and just return _biggest_ one?

See: https://pkg.go.dev/golang.org/x/mod/semver#Sort